File: src/python/open_redirect.py
"""CWE-601: Open redirect."""

from flask import Flask, request, redirect

app = Flask(__name__)


@app.route("/login")
def login():
    next_url = request.args.get("next", "/")
    return redirect(next_url)


@app.route("/out")
def out():
    return redirect(request.args.get("url"))


if __name__ == "__main__":
    app.run(port=4008)

File: src/javascript/open-redirect.js
// CWE-601: Open redirect.

const express = require('express');
const app = express();

app.get('/login', (req, res) => {
  const next = req.query.next || '/';
  res.redirect(next);
});

app.get('/out', (req, res) => {
  res.writeHead(302, { Location: req.query.url });
  res.end();
});

app.listen(3008);
